### PR TITLE
parse static function definitions

### PR DESCRIPTION
**Pull request by colleend**
for https://github.com/returntocorp/semgrep/issues/1809

### CODE REVIEW

File: lang_php/parsing/parser_php.mly [hunk 1]
```diff
@@ -544,6 +544,7 @@ function_body:
 async_opt:
```
Comment (aryx) [inline lang_php/parsing/parser_php.mly:544]: I would probably have renamed this to attribute_opt now that it's not just async.

File: tests/php/parsing_errors/static_function.php [hunk 2]
```diff
@@ -0,0 +1,7 @@
+<?php
```
Comment (aryx) [inline tests/php/parsing_errors/static_function.php:1]: again, this should be in parsing/

**Review by aryx — APPROVED** [review]
Hopefully this does not introduce s/r conflicts.
I'll fix CI and move the test file to the appropriate place.

**Comment by aryx** [comment]
argh, 2 s/r conflicts. I'll look.

**Comment by aryx** [comment]
Also @colleend it would be better if in your commit message you have a Test plan: section each time, explaining how
you tested your improvment. For example here, the command to parse the test file
and the output showing it parses it correctly.
Of course if you put the test under parsing/, you can just say 'make test' because it is automatically tested for regression.

File: lang_php/parsing/parser_php.mly (hunk 1 above)
Comment (colleend) [reply lang_php/parsing/parser_php.mly:544]: ohhh yes good point!

**Comment by colleend** [comment]
ah gotcha! I'll put a test plan in there next time 😅 , I forgot that the test should be under parsing!

**Pull request merged**